Jenkinsfile

```diff
@@ -12,18 +14,20 @@ def runTests(name, bedrockCommand, runAll) {
   }
 
   def command = runAll ? bedrockCommand + ' --ignore-lerna-changed=true' : bedrockCommand
-  def successfulTests = execHandle(command)
+  def testStatus = exec(script: command, returnStatus: true)
 
-  echo "Writing JUnit results for " + name + " on node: $NODE_NAME"
+  echo "Writing JUnit results for ${name} on node: $NODE_NAME"
   junit allowEmptyResults: true, testResults: 'scratch/TEST-*.xml'
 
-  if (!successfulTests) {
-    echo "Tests failed for " + name + " so passing failure as exit code for node: $NODE_NAME"
-    exec("exit 1")
+  // If the tests failed (exit code 4) then just mark it as unstable
+  if (testStatus == 4) {
```

Comment:
is `4` something we should have as a constant in waluigi somewhere or is that overkill?

Reply:
Sounds like overkill to me... if we did want something in Waluigi I'd abstract the entire function instead. That was something I considered, but it was likely this previously and isn't really in scope of just upgrading.

Reply:
I'll log a tech debt ticket and throw it in the editor team backlog for this, as it should probably be cleaned up. It needs a function similar to `bedrockTests` where it allows a command to be passed in instead basically, as that's why it can't be used atm.

Reply:
Logged as TINY-9313.